jcl/src/java.base/share/classes/jdk/internal/misc/Unsafe.java

```diff
@@ -0,0 +1,6138 @@
+/*[INCLUDE-IF Sidecar19-SE-B174]*/
+/*******************************************************************************
+ * Copyright (c) 2017, 2017 IBM Corp. and others
+ *
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/
+ * or the Apache License, Version 2.0 which accompanies this distribution and
+ * is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * This Source Code is also Distributed under one or more Secondary Licenses,
+ * as those terms are defined by the Eclipse Public License, v. 2.0: GNU
+ * General Public License, version 2 with the GNU Classpath Exception [1]
+ * and GNU General Public License, version 2 with the OpenJDK Assembly
+ * Exception [2].
+ *
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] http://openjdk.java.net/legal/assembly-exception.html
+ *******************************************************************************/
+package jdk.internal.misc;
+
+import java.lang.Class;
+import java.lang.ClassLoader;
+import java.lang.reflect.Field;
+import java.security.ProtectionDomain;
+import java.util.Objects;
+
+public final class Unsafe {
+
+	/* unsafe instance */
+	private static final Unsafe theUnsafe;
+
+	/**
+	 * Represents an invalid field offset value
+	 */
+	public static final int INVALID_FIELD_OFFSET;
+	
+	/**
+	 * Starting offset of Boolean array
```

Comment:
I don't think any of the native type names should be capitalized, as they could be confused with objects like java.lang.Boolean.